### PR TITLE
Rubocop fixes and bundle updates

### PR DESCRIPTION
**Pull request by jscodefix**

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -4,10 +4,17 @@
 # https://github.com/bbatsov/rubocop/blob/master/config/enabled.yml
 # https://github.com/bbatsov/rubocop/blob/master/config/disabled.yml
 
-Rails:
-  Enabled: true
```
Comment (aduth) [inline .rubocop.yml:8]: On a similar point to previous, were we relying on any Rails default cops?

File: .rubocop.yml [hunk 2]
```diff
@@ -4,10 +4,17 @@
...
-
-Layout/AlignParameters:
+AllCops:
+  Exclude:
+    - 'node_modules/**/*'
+    - 'public/**/*'
+  TargetRubyVersion: 3.0.4
+  UseCache: true
+  DisabledByDefault: true
```
[the remark this answers is not in the document]
Comment (jscodefix) [reply .rubocop.yml:13]: My approach to these changes is to bring more currency to Rubocop and the gems; without breaking any functionality. This in preparation for a forthcoming addition for inherited proofing.

Regarding the sharing of a .rubocop.yml, not sure it makes (total) sense here; the primary reason being, this is a Sinatra app, so I thought removing the Rails cop was good; not completely sure about that. Otherwise, sure sharing the same cops may make sense; but how? (https://github.com/percy/percy-style)

File: .rubocop.yml (hunk 1 above)
Comment (jscodefix) [reply .rubocop.yml:8]: ... continuing from my Rails comment above, there are 2 requires to _active_support_ (though I don't see any method calls). Then, adding Rails cop back would make sense, which could mean sharing the .rubocop.yml could make sense; but how?

File: .rubocop.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (aduth) [reply .rubocop.yml:13]: > My approach to these changes is to bring more currency to Rubocop and the gems; without breaking any functionality. This in preparation for a forthcoming addition for inherited proofing.

But changing the `DisabledByDefault` would be a breaking change, since we'd no longer be enforcing any of the rules which we were previously enabled by default, right?

> Regarding the sharing of a .rubocop.yml, not sure it makes (total) sense here; the primary reason being, this is a Sinatra app, so I thought removing the Rails cop was good;

Ah, I had a brain fart and overlooked the fact that this is not a Rails application. That makes sense.

>Otherwise, sure sharing the same cops may make sense; but how? (https://github.com/percy/percy-style)

Yeah, not totally sure about this in the Ruby/Rubocop landscape. For JavaScript, we have a [shared ESLint plugin](http://npmjs.com/package/@18f/eslint-plugin-identity) that's published to NPM, and it includes optional support for different configurations based on the project's needs (which could be a good way to separate out the Rails-specific configuration, if a similar convention would exist in a published Gem configuration).

**Comment by jscodefix** [comment]
OK, Andrew, per your good points above, I made these simple adjustments:
> EnabledByDefault: true
> NewCops: disable
Now, I'm second-guessing my "update the gems, cuz this is a small repo" idea. The biggest issue is that Rubocop went to the latest version (1.36.0) so the YML config keys changed, which makes it out of sync with `identity-idp`.  If you think this is OK, then I will proceed, otherwise I will cancel this PR. Please advise.  (Personally, I could manage to differences; I don't foresee major issues with the Ruby cop differences. But, I don't have much history with Rubocop.)

**Comment by aduth** [comment]
> The biggest issue is that Rubocop went to the latest version (1.36.0) so the YML config keys changed, which makes it out of sync with `identity-idp`.

I'm not quite sure I follow. Is this in reference to the renamed configuration names in Rubocop? I can see how it'd be nice to bring those up to date, yes.

**Review by aduth — APPROVED** [review]
I still think it'd be nice to have a common configuration, but yes, I think this is a good incremental step to get us up to date with the current configuration 👍

File: lib/tasks/write_deploy_json.rake [hunk 3]
```diff
@@ -1,10 +1,11 @@
 namespace :login do
   desc 'generate a generic deploy.json file'
-  task :deploy_json do
+  task deploy_json: :environment do
```
Comment (aduth) [inline lib/tasks/write_deploy_json.rake:3]: This seems to be causing issues for the build. What was it added for?

**Review by aduth — CHANGES_REQUESTED** [review]

File: app.rb [hunk 4]
```diff
@@ -228,12 +229,14 @@ def idp_public_key
     end
 
     def token(code)
+      logger.debug("DEBUG: in #token, token_endpoint = #{openid_configuration[:token_endpoint]}")
```
Comment (aduth) [inline app.rb:232]: Were we intending to add new logging in this pull request?

File: app.rb [hunk 5]
```diff
@@ -31,6 +31,7 @@ class OpenidConnectRelyingParty < Sinatra::Base
 
     configure :development do
       require 'byebug'
+      set :logging, Logger::DEBUG
```
Comment (aduth) [inline app.rb:34]: Were we intending to change the log level? Is this related to Rubocop?

File: lib/tasks/write_deploy_json.rake (hunk 3 above)
Comment (jscodefix) [reply lib/tasks/write_deploy_json.rake:3]: Cuz Rubocop wants that.  I'm working on it.  Thanks for the continuous attention; apologies for not pushing faster... multitasking.

File: app.rb (hunk 4 above)
Comment (jscodefix) [reply app.rb:232]: That is an oversight; removing.  Thank you and again, apologies for not updating faster... my workflow should improve.

File: app.rb (hunk 5 above)
Comment (jscodefix) [reply app.rb:34]: I was going to leave that in, for development purposes.  I'll just remove it, since I can simply re-add when I want debug output.  Personally, I would prefer to have debug/info outputs for this app, since it is used strictly for testing.

Comment (aduth) [reply app.rb:29]: > it is used strictly for testing.

These apps are [deployed](https://handbook.login.gov/articles/appdev-cloud-gov-deploy.html), where I don't think we'd want this level of logging.

File: app.rb (hunk 4 above)
Comment (aduth) [reply app.rb]: > apologies for not updating faster... my workflow should improve.

No need to apologize! It certainly isn't my intent to rush any changes here.

File: app.rb (hunk 5 above)
Comment (jscodefix) [reply app.rb:29]: OK, deployed for testing. So, I added it under the :development configuration, along with byebug, and thought it would be useful for other developers.  I'll remove it.

I can't tell what other things need to be resolved, to have this PR approved.

File: app.rb [hunk 6]
```diff
@@ -1,8 +1,6 @@
 # frozen_string_literal: true
 
 require 'dotenv/load'
-require 'active_support/core_ext/hash/indifferent_access'
-require 'active_support/core_ext/object/to_query'
```
Comment (aduth) [inline app.rb:5]: Were these changes related to Rubocop? I see that it still works without it, but I feel like this leaves us with an implicit assumption that one of our other dependencies will be making these methods available (`json-jwt`), which could easily break in the future. I think having it explicitly required like it had been was a good idea.

**Review by aduth — APPROVED** [review]

File: app.rb (hunk 6 above)
Comment (jscodefix) [reply app.rb:5]: I see your point; I'll investigate requiring a gem (json/jwt) and its dependencies. Also, I see there could be a missing test; as you point out, would then break app.  Thanks for approving.  Let me gain more confidence in removing those 2 require lines.
